google_guest_agent/main.go

```diff
@@ -175,6 +176,9 @@ func runAgent(ctx context.Context) {
 
 	agentInit(ctx)
 
+	cmdsrv := command.Init(ctx)
```

Comment:
we want to protect this initialization with a feature flag, it'd look like this:

```
if cfg.Get().Unstable.CommandMonitorEnabled {
  cmdServer := command.Init(ctx)
  defer cmdServer.Close()
}
```

Reply:
My intention was to write something that didn't require an explicit enablement flag, as if some subsystem X depends on command monitor functionality this would allow it to start listening whenever X was enabled (and registered its handlers with the monitor) and otherwise do nothing. This was the same reason for the Start/Stop signals. We don't have a flag to disable the events system either, I was considering this as a similar system, but we can move to a feature flag gated one.

Reply:
I quite don't follow this `as if some subsystem X depends on command monitor functionality this would allow it to start listening whenever X was enabled`, what do you mean? Having a feature flag allows customers to disable it entirely if they need/want, if we can't allow it then we are doing something wrong and should fix. PS: Initially we want to keep it disabled by default until we launch the feature.

Reply:
I mean, imagine this configuration for some future subsystem of guest-agent that depends on command-monitor:

```
[X]
enabled = true
[CommandMonitor]
enabled = false
```

If X needs the command monitor for core functionality, how should it handle it? Just log an error? Disable itself? Should subsystems never design around guaranteed availability of the command monitor? We would need to document this, or at least document that subsystems should direct users to enable command monitor if it's required.

The intention for my current design is that if X is the only user of command monitor, then it will open the socket when `X.Enabled` is true and will not open it when `X.Enabled` is false.

Reply:
If subsystem X exposes a command and command monitor is disabled then the `Register()` call will be a no-op - yes, we can log that it resulted in a no-op.

Subsystem X can expose its own configuration section and expose a `enabled` flag, from a user perspective it's a "inclusive or" where Subsystem X and command monitor enabled flags must be true to enable it.

In short, a command monitor enabled flag set to false will disable all commands.

Reply:
Ack. I'll include some documentation for this.